### PR TITLE
Add accentColor props for iOS datetimepicker (#20)

### PR DESCRIPTION
**Pull request by simonnagl**
# Summary

This adds prop accentColor for iOS.
This relates to #20.

### What's required for testing (prerequisites)?

Testing can be done with the example app. I added another input (accentColor) which can be used for testing.

### What are the steps to reproduce (after prerequisites)?

Select display to compact or inline.
Enter a color value (i.e. `#f00`) into the `accentColor` field.
Check if picker appearance changes.

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ❌     |

## Checklist

- [x] I have tested this on a simulator
- [x] I added the documentation in `README.md`
- [x] I updated the typed files (TS and Flow)
- [x] I added a sample use of the API in the example project (`example/App.js`)
- [ ] I have added automated tests, either in JS or e2e tests, as applicable

e2e test can not check color changes.

### CODE REVIEW

**Comment by peteratdivvy** [comment]
Would absolutely love for this to land. We really need a way to set the tintColor for the calendar.

File: example/App.js [hunk 1]
```diff
@@ -131,7 +134,12 @@ export const App = () => {
     return (
       <SafeAreaView style={[backgroundStyle, {flex: 1}]}>
         <StatusBar barStyle="dark-content" />
-        <ScrollView testID="DateTimePickerScrollView">
+        <ScrollView
+          testID="DateTimePickerScrollView"
+          ref={scrollRef}
+          onContentSizeChange={() =>
+            scrollRef.current?.scrollToEnd({animated: true})
```
Comment (vonovak) [inline example/App.js:141]: I'm afraid this is breaking the e2e tests on android as some ui elements are not visible. We need to get a testing emulator with a bigger screen I guess, do you want to take a look at this?

thanks

File: example/e2e/utils/matchers.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ const elementByText = (text) => element(by.text(text));
 const getDateTimePickerIOS = () =>
   element(by.type('UIPickerView').withAncestor(by.id('dateTimePicker')));
 
-const getInlineTimePickerIOS = () => element(by.label('Time Picker'));
+const getInlineTimePickerIOS = () => element(by.type('RNDateTimePicker'));
```
Comment (vonovak) [inline example/e2e/utils/matchers.js:9]: just wondering, why this change?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -34,6 +34,7 @@ For cleaning all the detox builds just run `npm run detox:clean`.
   ```sh
   # Debug requires to run Metro Bundler
   yarn start
+  cd "example/ios" && npx pod-install && cd "../.."
```
Comment (vonovak) [inline CONTRIBUTING.md:37]:
```suggestion
  cd "example/ios" && npx pod-install && cd -
```

same thing but less typing 😄

**Review by vonovak — COMMENTED** [review]
thank you for the PR!

it's looking good, we just need to get the e2e tests green. Let me know if you want to have a look or I can do too 👍

File: example/e2e/utils/matchers.js (hunk 2 above)
Comment (simonnagl) [reply example/e2e/utils/matchers.js:9]: I think we can revert this. This was needed before 93278605fcd14bdb735b015e9f725f2c21469e23. The additional `<View>` Tag got the label `Time Picker`. Detox found two elements with the same label and failed.

File: example/App.js (hunk 1 above)
Comment (simonnagl) [reply example/App.js:141]: I will take a look on this.

Comment (simonnagl) [reply example/App.js:141]: Just used a bigger device. Now all steps are green.

File: example/e2e/utils/matchers.js (hunk 2 above)
Comment (simonnagl) [reply example/e2e/utils/matchers.js:9]:
```suggestion
const getInlineTimePickerIOS = () => element(by.label('Time Picker'));
```

File: example/App.js (hunk 1 above)
Comment (vonovak) [reply example/App.js:141]: we probably don't need scrollToEnd in onContentSizeChange (sounds like a weird behavior), so I guess onContentSizeChange can be removed?

File: README.md [hunk 4]
```diff
@@ -387,6 +388,11 @@ Allows changing of the textColor of the date picker. Has effect only when `displ
 <RNDateTimePicker textColor="red" />
 ```
 
+#### `accentColor` (`optional`, `iOs only`)
+
+Allows changing the accentColor (tintColor) of the date picker.
```
Comment (vonovak) [inline README.md:393]: I'm wondering, why did you choose to name this `accentColor` and not `tintColor`? Thanks

File: package.json [hunk 5]
```diff
@@ -114,11 +114,11 @@
         "build": "export RCT_NO_LAUNCH_PACKAGER=true && (cd example/android && ./gradlew assembleDebug assembleAndroidTest -DtestBuildType=debug)",
         "type": "android.emulator",
         "device": {
-          "avdName": "Pixel_4_Android_12_api_31"
+          "avdName": "Pixel_5_Android_12_api_31"
         }
       },
       "android.device.debug": {
-        "binaryPath": "example/android/app/build/outputs/apk/debug/app-debug.apk",
+        "binaryPath": "build/outputs/apk/androidTest/debug/datetimepicker-debug-androidTest.apk",
```
Comment (vonovak) [inline package.json:121]: again just wondering why this was needed? 🙂

File: example/App.js (hunk 1 above)
Comment (simonnagl) [reply example/App.js:141]: iOS depends on it at the moment. Also it is very useful for newcomers trying the example app on small devices. Before this change if you tap "Show picker" nothing could happen because the picker shows up at the bottom outside of the viewport.
Now, if the picker gets rendered or changed, content Size Changes and the app shows the rendered picker because it scrolls down.
This could also be done not `onContentSizeChange` of the scroll view, but after the picker rendered. Then we would need to wrap the picker component in another component firing an callback which lets the scollView scroll.

I thought this might be to much work for an example app. The solution would be more descriptive than reacting to `onContentSizeChange`.

I will install the example app into an small android simulator and test if the behaviour with scroll to end also works good for android. If not i will limit it to iOS.

Please tell me if the solution like now is ok, or I should bind the scrollToDatepicker more directly to changing the date picker visibility or props.

File: README.md (hunk 4 above)
Comment (simonnagl) [reply README.md:393]: I think `tintColor` is very `iOS` specific. datetimepicker might support setting an `accentColor` for android too, in the future.

I thought about three options:
* `tintColor` -> Good for people who are familiar with iOS programming, maybe not descriptive for the rest.
* `primaryColor` -> Good for most people, but maybe misleading without `secondaryColor` 
* `accentColor` -> I find `accentColor` very descriptive.

Please let me now if you want to change the API.

File: package.json (hunk 5 above)
Comment (simonnagl) [reply package.json:121]: This was committed per accident:

```suggestion
        "binaryPath": "example/android/app/build/outputs/apk/debug/app-debug.apk",```

File: README.md [hunk 6]
```diff
@@ -387,6 +388,11 @@ Allows changing of the textColor of the date picker. Has effect only when `displ
 <RNDateTimePicker textColor="red" />
 ```
 
+#### `accentColor` (`optional`, `iOs only`)
```
Comment (vonovak) [inline README.md:391]:
```suggestion
#### `accentColor` (`optional`, `iOS only`)
```

File: README.md [hunk 7]
```diff
@@ -387,6 +388,11 @@ Allows changing of the textColor of the date picker. Has effect only when `displ
...
+
+Allows changing the accentColor (tintColor) of the date picker.
+Has not effect when `display` is `"spinner"`.
```
Comment (vonovak) [inline README.md:394]:
```suggestion
Has no effect when `display` is `"spinner"`.
```

**Review by vonovak — APPROVED** [review]

**Comment by simonnagl** [comment]
@vonovak Thank you a lot for maintaining this repo! And for the quick an gentle code review.

I may create further pull requests if needed personally. I likely will not work on the project on a open issue basis.
